### PR TITLE
[CPM-161] Add routing to courses page

### PR DESCRIPTION
**Pull request by daltonlim**
## Changes

- Course routing has been added to the courses page
- /courses/new navigates to adding a new course
- /courses/id navigates to a given program
- /courses/ and invalid ids get redirected to the create or select a course page

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-161]

Connects #y

[CPM-161]: https://momentum-agile.atlassian.net/browse/CPM-161

### CODE REVIEW

File: client/src/pages/Courses/EmpyCourse.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Flex, Text } from "@chakra-ui/core";
+import { colors as c } from "../../colors";
+
+const EmptyCourse = () => {
```
Comment (vee-christella) [inline client/src/pages/Courses/EmpyCourse.js:5]: Can we rename this file name to `EmptyCourse` (not Empy hehe)

**Review by vee-christella — APPROVED** [review]
Thanks for doing this! 😄

File: client/src/pages/Courses/EmpyCourse.js (hunk 1 above)
Comment (daltonlim) [reply client/src/pages/Courses/EmpyCourse.js]: done

File: client/src/pages/Courses/Courses.js [hunk 2]
```diff
@@ -17,11 +19,17 @@ const Courses = () => {
...
+    const { courseId } = useParams();
 
     useEffect(() => {
         setSelectedCourse(isAddingCourse ? {} : data[currRow] || {});
         setIsEditing(false);
-    }, [data, currRow, isAddingCourse]);
+        const rowID = data.findIndex((course) => course._id === courseId);
+        setCurrRow(rowID.toString());
+        setIsAddingCourse(courseId === "new");
+        console.log(rowID);
```
Comment (FocalChord) [inline client/src/pages/Courses/Courses.js:31]: pls remove : )

Comment (daltonlim) [reply client/src/pages/Courses/Courses.js:31]: my bad. thought the checker had a console log checker

File: client/src/pages/Courses/Courses.js [hunk 3]
```diff
@@ -17,11 +19,16 @@ const Courses = () => {
...
+    const { courseId } = useParams();
 
     useEffect(() => {
         setSelectedCourse(isAddingCourse ? {} : data[currRow] || {});
         setIsEditing(false);
-    }, [data, currRow, isAddingCourse]);
+        const rowID = data.findIndex((course) => course._id === courseId);
+        setCurrRow(rowID.toString());
+        setIsAddingCourse(courseId === "new");
+    }, [data, currRow, isAddingCourse, courseId]);
```
Comment (FocalChord) [inline client/src/pages/Courses/Courses.js:31]: This useEffect seems to have many dependencies, could we split your `courseId` into a new useEffect to seperate concerns?

**Review by FocalChord — APPROVED** [review]
lgtm

**Pull request merged**